addon/PreviewProvider.js

```diff
@@ -192,7 +192,7 @@ PreviewProvider.prototype = {
     }
 
     return this._asyncGetEnhancedLinks(processedLinks, event).then(
-      cachedLinks => this._getFaviconColors(cachedLinks));
+      cachedLinks => this._getFaviconColors(cachedLinks)).catch(err => Cu.reportError(err));
```

Comment:
I think if you `.catch()` a promise error, it should then resume onto the next chained `.then()`, IIRC.

So in places where we return a promise, not sure if we'd want a catch + log + throw action, so the usage side can handle any errors as well.